### PR TITLE
docs: storage.md: clarify storage.tsdb.retention.time description

### PR DESCRIPTION
**Pull request by tesla59**
Clarify the content of storage.md, specifically `storage.tsdb.retention.time`  flag. 
mentioned here: https://github.com/prometheus/prometheus/issues/13967#issuecomment-2073040519
https://github.com/prometheus/prometheus/blob/76b0318ed52e655e96a3a4734e3678bf55801616/cmd/prometheus/main.go#L345-L346

### CODE REVIEW

File: docs/storage.md [hunk 1]
```diff
@@ -84,8 +84,10 @@ or 31 days, whichever is smaller.
 Prometheus has several flags that configure local storage. The most important are:
 
 - `--storage.tsdb.path`: Where Prometheus writes its database. Defaults to `data/`.
-- `--storage.tsdb.retention.time`: When to remove old data. Defaults to `15d`.
-  Overrides `storage.tsdb.retention` if this flag is set to anything other than default.
+- `--storage.tsdb.retention.time`: How long to retain samples in storage. When this flag is
+  set it overrides `storage.tsdb.retention`. If neither this flag nor `storage.tsdb.retention`
```
Comment (machine424) [inline docs/storage.md:88]:
```suggestion
  set, it overrides `storage.tsdb.retention`. If neither this flag nor `storage.tsdb.retention`
```

File: docs/storage.md [hunk 2]
```diff
@@ -84,8 +84,10 @@ or 31 days, whichever is smaller.
...
+  nor `storage.tsdb.retention.size` is set, the retention time defaults to `15d`.
+  Units Supported: y, w, d, h, m, s, ms.
```
Comment (machine424) [inline docs/storage.md:90]:
```suggestion
  Supported units: y, w, d, h, m, s, ms.
```

**Review by machine424 — COMMENTED** [review]
just some nits.
Thanks.

**Comment by tesla59** [comment]
Hi @machine424 
Appreciate your review.
Made the suggested changes
Thanks

**Review by machine424 — APPROVED** [review]
lgtm

**Pull request merged**